### PR TITLE
ci: allow to not truncate log file

### PR DESCRIPTION
**Pull request by hicqu**
Signed-off-by: qupeng <qupeng@pingcap.com>

### What problem does this PR solve?

When running one test case, for example:
```bash
cargo test --all test_node_merge_transfer_leader
```
Both `integrations-*` and `failpoints-` will be runned, which will open and truncate the target log file twice.

### What is changed and how it works?

Add a new environment variable `LOG_APPEND`.
Setting it to any value will use append instead of truncate when opening log file.

### Check List 

Tests 

- No code

Side effects

- Performance regression
    - Consumes more CPU
    - Consumes more MEM
- Breaking backward compatibility

### Release note 
* No release note.

### CODE REVIEW

File: components/test_util/src/logging.rs [hunk 1]
```diff
@@ -100,7 +100,19 @@ pub fn init_log_for_test() {
             &env::var("LOG_LEVEL").unwrap_or_else(|_| "debug".to_owned()),
         )
         .unwrap();
-        let writer = output.map(|f| Mutex::new(File::create(f).unwrap()));
+        let writer = if env::var("LOG_APPEND").is_ok() {
+            output.map(|f| {
+                Mutex::new(
+                    OpenOptions::new()
+                        .create(true)
+                        .append(true)
```
Comment (BusyJay) [inline components/test_util/src/logging.rs:108]: How about replacing `true` with `env::var("LOG_APPEND").is_ok()`?

**Review by BusyJay — APPROVED** [review]

**Review by Fullstop000 — APPROVED** [review]
It seems this is relate to the failing tests

**Pull request merged**